cm/option.go

```diff
@@ -8,6 +12,27 @@ type Option[T any] struct {
 	option[T]
 }
 
+// MarshalJSON implements the json.Marshaler interface for [Option].
+func (o Option[T]) MarshalJSON() ([]byte, error) {
+	return json.Marshal(o.Some())
+}
+
+// UnmarshalJSON unmarshals the Option from JSON.
+func (o *Option[T]) UnmarshalJSON(buf []byte) error {
+	if len(buf) == 0 {
```

Comment:
This should also check if buf == `null`.

Reply:
👍  the `null` story here is interesting. will push it up as part of the breaking this PR into many